modules/account-lib/test/unit/coin/sol/transactionBuilder/stakingActivateBuilder.ts

```diff
@@ -0,0 +1,275 @@
+
+      it('build when stakingAddress is not provided', async () => {
+        const txBuilder = factory.getStakingActivateBuilder();
+        txBuilder.sender(wallet.pub);
+        txBuilder.amount(amount);
+        txBuilder.validator(validator.pub);
+        txBuilder.nonce(recentBlockHash);
+        txBuilder.sign({ key: wallet.prv });
+        await txBuilder.build().should.rejectedWith('stakingAddress must be set before building the transaction');
+      });
+
+      it('build when validator is not provided', async () => {
+        const txBuilder = factory.getStakingActivateBuilder();
+        txBuilder.stakingAddress(stakeAccount.pub);
+        txBuilder.sender(wallet.pub);
+        txBuilder.amount(amount);
+        txBuilder.nonce(recentBlockHash);
+        txBuilder.sign({ key: wallet.prv });
+        await txBuilder.build().should.rejectedWith('Validator must be set before building the transaction');
+      });
+
+      it('to sign twice with the same key', () => {
+        const txBuilder = factory.from(testData.STAKING_ACTIVATE_UNSIGNED_TX);
+        txBuilder.sign({ key: wallet.prv });
+        should(() => txBuilder.sign({ key: wallet.prv })).throwError('Duplicated signer: ' + wallet.prv?.toString());
+      });
+    });
+
+    describe('From and sign', () => {
+      describe('Succeed', () => {
+        it('build from a unsigned staking activate and sign it', async () => {
+          const txBuilder = factory.from(testData.STAKING_ACTIVATE_UNSIGNED_TX);
+
+          txBuilder.sign({ key: wallet.prv });
+          txBuilder.sign({ key: stakeAccount.prv });
+          const tx = await txBuilder.build();
+          tx.inputs.length.should.equal(1);
+          tx.inputs[0].should.deepEqual({
+            address: wallet.pub,
+            value: amount,
+            coin: 'tsol',
+          });
+          tx.outputs.length.should.equal(0);
+          const rawTx = tx.toBroadcastFormat();
+          should.equal(Utils.isValidRawTransaction(rawTx), true);
+          should.equal(rawTx, testData.STAKING_ACTIVATE_SIGNED_TX);
+        });
+
+        it('build from a unsigned staking activate with memo and sign it', async () => {
+          const txBuilder = factory.from(testData.STAKING_ACTIVATE_UNSIGNED_TX_WITH_MEMO);
+          txBuilder.sign({ key: wallet.prv });
+          txBuilder.sign({ key: stakeAccount.prv });
+          const tx = await txBuilder.build();
+          tx.inputs.length.should.equal(1);
+          tx.inputs[0].should.deepEqual({
+            address: wallet.pub,
+            value: amount,
+            coin: 'tsol',
+          });
+          tx.outputs.length.should.equal(0);
+          const rawTx = tx.toBroadcastFormat();
+          should.equal(Utils.isValidRawTransaction(rawTx), true);
+          should.equal(rawTx, testData.STAKING_ACTIVATE_SIGNED_TX_WITH_MEMO);
+        });
+      });
+
+      describe('Fail', () => {
+        it('build from a unsigned staking activate and fail to sign it', async () => {
```

Comment:
build from an unsigned staking activate and fail to sign it

Reply:
changed locally, waiting for other comments to upload this.